### DISCUSSION TITLE
Are props returned by getStaticProps during rehydration?

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [vercel/next.js#21087] [Help] [answered]
Let's say I have the following getStaticProps function:

```ts
export const getStaticProps: GetStaticProps = () => {
    // pretend like this is coming from a database
    const pageData = {
        itemOne: 'this can be public',
        itemTwo: 'this should be kept secret'
    }
    // getStaticProps expects a promise
    return Promise.resolve({ props: { pageData } })
}
```

Then NextJS will create the static page.

Will a malicious user be able to get the value of `pageData.itemTwo` even if the page doesn't use that variable? In other words, `pageData.itemTwo` is never used on the actual page, but will it be available in rehydration? Then someone could theoretically get the value of `pageData.itemTwo`.

If so, the easy solution would be to just not include it in the output of getStaticProps, but just want to know if that's necessary.

**Comment by jamesmosier** [comment, marked as answer]
Yes, anything returned in `getStaticProps` is exposed to the client. Look in the DOM for `id="__NEXT_DATA__"` and you'll see all the props from `getStaticProps` there. React uses those during rehydration to render the components in the browser.
